### PR TITLE
Add users module

### PR DESCRIPTION
**Pull request by hunter-s70**
Add users module with user model. It's a preparations for future authorization

### CODE REVIEW

File: src/users/user.service.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Injectable } from '@nestjs/common';
+import { User } from './user.entity';
+
+@Injectable()
+export class UserService {
+  async findAll(): Promise<User[]> {
+    return User.find();
```
Comment (haltaction) [inline src/users/user.service.ts]: Better to use repositories for all work with Database.

https://docs.nestjs.com/techniques/database#repository-pattern
https://ru.wikipedia.org/wiki/%D0%A0%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5_%D0%BE%D1%82%D0%B2%D0%B5%D1%82%D1%81%D1%82%D0%B2%D0%B5%D0%BD%D0%BD%D0%BE%D1%81%D1%82%D0%B8

File: src/app.module.ts [hunk 2]
```diff
@@ -2,10 +2,29 @@ require('dotenv').config();
...
+      synchronize: true,
+    }),
+    UserModule
+  ],
   controllers: [AppController],
   providers: [AppService],
 })
-export class AppModule {}
+export class AppModule {
+  constructor(private connection: Connection) {}
```
Comment (hunter-s70) [inline src/app.module.ts:29]: @haltaction I'm not sure if it's needed. It was in example, but I'm also using `TypeOrmModule.forFeature([User])` into the UsersModule

File: src/app.module.ts [hunk 3]
```diff
@@ -2,10 +2,29 @@ require('dotenv').config();
...
-  imports: [],
+  imports: [
+    TypeOrmModule.forRoot({
+      type: 'postgres',
+      host: process.env.DB_HOST,
+      port: Number(process.env.DB_PORT),
+      username: process.env.DB_USER,
+      password: process.env.DB_PASS,
+      database: process.env.DB_NAME,
+      entities: [User],
```
Comment (hunter-s70) [inline src/app.module.ts:20]: @haltaction we have to add every entity here, isn't it?

File: src/users/user.service.ts (hunk 1 above)
Comment (hunter-s70) [reply src/users/user.service.ts]: @haltaction fixed

File: src/app.module.ts (hunk 3 above)
Comment (haltaction) [reply src/app.module.ts:20]: For now yes, it's fine. In the future, this block could be moved to a separate module, e.g. _DatabaseModule_ 
And in _app.module_ we could only import this module.

File: src/app.module.ts (hunk 2 above)
Comment (haltaction) [reply src/app.module.ts:29]: You could delete it/ If not sure, try to delete and watch what is happening :)

File: src/app.module.ts (hunk 3 above)
Comment (haltaction) [reply src/app.module.ts:20]: Also there is a way for many entities - use `"entities": ["dist/**/*.entity{.ts,.js}"],` https://docs.nestjs.com/techniques/database#typeorm-integration

**Review by haltaction — APPROVED** [review]

**Comment by hunter-s70** [comment]
It's better to keep all authorization changes in one PR https://github.com/hunter-s70/nest7-test-app/pull/4

**Pull request closed without merging**